### PR TITLE
Fix Machines Pages Buttons

### PR DESCRIPTION
**Pull request by Calvinp**
This fixes the buttons on the machines pages so that they actually work.

### CODE REVIEW

**Comment by Calvinp** [comment]
This is good from my end. @tpetr @kwm4385 @wolfd

File: SingularityUI/app/components/machines/MachinesPage.jsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import OldTable from '../common/OldTable';
 const MachinesPage = React.createClass({
```
Comment (wolfd) [inline SingularityUI/app/components/machines/MachinesPage.jsx]: We can make this a functional component

File: SingularityUI/app/components/machines/ModalButton.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import React, { PropTypes } from 'react';
+import FormModal from '../common/FormModal';
+import {OverlayTrigger, Tooltip} from 'react-bootstrap/lib';
+
+class ModalButton extends React.Component {
```
Comment (wolfd) [inline SingularityUI/app/components/machines/ModalButton.jsx]: I've been using this pattern for [Modal](https://github.com/HubSpot/Singularity/blob/8ee68b81edb0cdc749e03f9de46ddca91798d3f4/SingularityUI/app/components/requests/BounceModal.jsx)s and [Button](https://github.com/HubSpot/Singularity/blob/8ee68b81edb0cdc749e03f9de46ddca91798d3f4/SingularityUI/app/components/requests/BounceButton.jsx)s

File: SingularityUI/app/components/machines/ModalButton.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React, { PropTypes } from 'react';
+import FormModal from '../common/FormModal';
+import {OverlayTrigger, Tooltip} from 'react-bootstrap/lib';
```
Comment (wolfd) [inline SingularityUI/app/components/machines/ModalButton.jsx]: is `/lib` necessary?

File: SingularityUI/app/components/machines/Racks.jsx [hunk 4]
```diff
@@ -2,10 +2,11 @@ import React, {PropTypes} from 'react';
 import MachinesPage from './MachinesPage';
 import PlainText from '../common/atomicDisplayItems/PlainText';
 import TimeStamp from '../common/atomicDisplayItems/TimeStamp';
-import Link from '../common/atomicDisplayItems/Link';
-import Glyphicon from '../common/atomicDisplayItems/Glyphicon';
+import Glyphicon from 'react-bootstrap/lib/Glyphicon';
```
Comment (wolfd) [inline SingularityUI/app/components/machines/Racks.jsx]: `import { Glyphicon } from 'react-bootstrap';`

File: SingularityUI/app/components/machines/Racks.jsx [hunk 5]
```diff
@@ -51,77 +61,48 @@ const Racks = React.createClass({
...
-        }}
-      />
-      );
-    }
-    return null;
+    return (__in__(rack.currentState.state, ['DECOMMISSIONING', 'DECOMMISSIONED', 'STARTING_DECOMMISSION']) &&
+      <ModalButton
+        buttonChildren={<Glyphicon glyph="new-window" />}
+        action="Reactivate Rack"
+        onConfirm={(data) => this.props.clear().then(() => this.props.reactivateRack(rack, data.message))}
```
Comment (wolfd) [inline SingularityUI/app/components/machines/Racks.jsx]: this is kinda long for a function in a prop

File: SingularityUI/app/components/machines/Racks.jsx [hunk 6]
```diff
@@ -221,15 +202,43 @@ const Racks = React.createClass({
...
       header = "Racks"
       states = {this.getStates()}
+      error = {this.props.error}
     />
     );
   }
 });
 
+function getErrorFromState(state) {
+  if (state.api.decommissionRack.error) {
```
Comment (wolfd) [inline SingularityUI/app/components/machines/Racks.jsx]: a `const { decommissionRack, removeRack, reactivateRack } = state.api;` might be nice here

**Comment by Calvinp** [comment]
I took the opportunity to replace the `OldTable` component with `SimpleTable`. Along with #1140, this almost fully removes that component.

**Comment by tpetr** [comment]
LGTM

**Pull request merged**